src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -613,6 +613,7 @@ bool checkSymbol(Symbol sym, TextSpan memberSpan, SymbolKind kind, out Symbol re
                             {
                                 MethodSymbol method => (Symbol)method.PartialImplementationPart,
                                 SourcePropertySymbol property => property.PartialImplementationPart,
+                                SourceEventSymbol ev => ev.PartialImplementationPart,
```

Comment:
I wonder if this should be a helper. Feel free to extract if you find there are several consistent enough usages like this one.

Also, the comment that `member.Locations` includes both parts is not true, feel free to delete it.